File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lp=0;
        int rp = height.size()-1;
        int ans=0;
        int currwater=0;
        while(lp<rp){
            int w=rp-lp;
            int h=min(height[lp],height[rp]);
            currwater=w*h;
            ans=max(ans,currwater);
            height[lp]<height[rp] ? lp++ : rp--;

        }
        return ans;

        
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] nums) {
        int n = nums.length;
        int st = 0;
        int end = n-1;
        int area = 1;
        int ans = Integer.MIN_VALUE;
        while(st<end){
            int b = end - st;
            int len = Math.min(nums[st],nums[end]);
            if(nums[st]<nums[end]){
            area = len * b;
            ans = Math.max(ans,area);
            st++;
            }else{
                area = len * b;
                ans = Math.max(ans,area);
                end--;
            }

        }
        return ans;
        
    }
}